src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -324,7 +324,7 @@ private void EnsureIOInitialized()
             // If at this point request was not upgraded just start a normal IO engine
             if (AsyncIO == null)
             {
-                AsyncIO = new AsyncIOEngine(_contextLock, _requestNativeHandle);
+                AsyncIO = new AsyncIOEngine(this, _requestNativeHandle);
```

Comment:
If the `IISHttpContext` is only used for its lock object, shouldn't we just pass the lock object? That way _contextLock doesn't need to be made internal either.

Reply:
It's used for https://github.com/dotnet/aspnetcore/pull/25883/files#diff-d8f0d56e181f0c0743395b9c9295093bR146